backend/routes/post.js

```diff
@@ -86,13 +72,14 @@ router.get('/:_pid/:_cid', (request, response, next) => {
 })
 
 // ADD NEW POST
-router.post('/', mid.requiresLogin, (request, response, next) => {
+router.post('/', mid.requiresLogin, async (request, response, next) => {
   let post = request.body
   post.author = mongoose.Types.ObjectId(request.session.userId)
+  const user = await User.getCurrentUser(request, response, next)
+  post.authorName = user.name
   controllerPost
     .addPost(post)
-    .then(async function(data) {
-      const user = await User.getCurrentUser(request, response, next)
+    .then(function(data) {
```

Comment:
good use of async/await, just to let you know. If you use async/await, you could convert the promises + then into await.

I was the one who wrote the code for the promises, as I didn't think I needed to add more code (I guess that is a lesson learned for me. To account for future possibilities of the addition of code into functions)

Reply:
I've converted the promises to async/await.